Framework/Geometry/src/Objects/ShapeFactory.cpp

```diff
@@ -198,6 +197,7 @@ boost::shared_ptr<Object> ShapeFactory::createShape(Poco::XML::Element *pElem) {
             idMatching[idFromUser] = parseCone(pE, primitives, l_id);
             numPrimitives++;
           } else if (!primitiveName.compare("hexahedron")) {
```

Comment:
This "hexahedron" string comparison is turing up a lot. You might consider creating a method for doing this repeated work for you somewhere.